src/trio/_tests/test_testing_raisesgroup.py

```diff
@@ -363,6 +612,23 @@ def test_matcher_tostring() -> None:
     )
 
 
+def test_raisesgroup_tostring() -> None:
+    def check_str_and_repr(s: str) -> None:
+        evaled = eval(s)
+        assert s == str(evaled) == repr(evaled)
+
+    check_str_and_repr("RaisesGroup(ValueError)")
+    check_str_and_repr("RaisesGroup(RaisesGroup(ValueError))")
+    check_str_and_repr("RaisesGroup(Matcher(ValueError))")
+    check_str_and_repr("RaisesGroup(ValueError, allow_unwrapped=True)")
+    check_str_and_repr("RaisesGroup(ValueError, match='aoeu')")
+
+    assert (
+        str(RaisesGroup(ValueError, match="[a-z]", check=bool))
+        == f"RaisesGroup(ValueError, match='[a-z]', check={bool!r})"
```

Comment:
Yeah, I'd definitely like the [nicerepr](https://github.com/HypothesisWorks/hypothesis/blob/690520e34d139628f86e108802a9a83cbca11fbb/hypothesis-python/src/hypothesis/internal/reflection.py#L461-L468), but that's a _lot_ of cursed support code... as above, monkeypatching might be the way to go.   Or, actually, we could define a cute proxy:

```python
def nicerepr(obj: object) -> str:
    global nicerepr
    if mod := sys.modules.get("hypothesis.internal.reflection"):
        # Not worth reimplementing, but if it's imported we might as well use it
        try:
            nicerepr = mod.nicerepr
            return nicerepr(obj)
        except Exception:
            nicerepr = repr  # maybe it moved?  don't try again.
    return repr(obj)

Reply:
given this is ultimately meant for pytest, making it monkeypatchable and having hypothesis's pytest plugin monkeypatch it would perhaps be neater?

Reply:
Yep, that sounds like a great way to do it.  And then for Trio we can monkeypatch it from Trio's _hypothesis_ plugin: https://github.com/python-trio/trio/blob/cfbbe2c1f96e93b19bc2577d2cab3f4fe2e81153/src/trio/_core/_run.py#L118-L123